File: jetbrains-plugin/src/main/kotlin/com/asakii/plugin/services/ClaudeSettingsService.kt
package com.asakii.plugin.services

import com.intellij.openapi.components.*
import com.intellij.openapi.project.Project
import kotlinx.serialization.Serializable

/**
 * Claude 设置服务
 * 
 * 管理 Claude 相关的配置（API Key、模型等）
 */
@State(
    name = "ClaudeCodePlusSettings",
    storages = [Storage("claudeCodePlus.xml")]
)
@Service(Service.Level.PROJECT)
class ClaudeSettingsService : PersistentStateComponent<ClaudeSettingsService.State> {
    
    @Serializable
    data class State(
        var apiKey: String? = null,
        var model: String = "claude-sonnet-4-5-20250929",
        var maxTokens: Int = 4096,
        var temperature: Double = 0.7,
        var maxTurns: Int = 10,
        var autoApproveTools: Boolean = false,
        var useDarkTheme: Boolean = true
    )
    
    data class Settings(
        val apiKey: String?,
        val model: String,
        val maxTokens: Int,
        val temperature: Double,
        val maxTurns: Int,
        val autoApproveTools: Boolean,
        val useDarkTheme: Boolean
    )
    
    private var state = State()
    
    /**
     * 获取当前设置
     */
    fun getSettings(): Settings {
        return Settings(
            apiKey = state.apiKey,
            model = state.model,
            maxTokens = state.maxTokens,
            temperature = state.temperature,
            maxTurns = state.maxTurns,
            autoApproveTools = state.autoApproveTools,
            useDarkTheme = state.useDarkTheme
        )
    }
    
    /**
     * 更新设置
     */
    fun updateSettings(settings: Settings) {
        state.apiKey = settings.apiKey
        state.model = settings.model
        state.maxTokens = settings.maxTokens
        state.temperature = settings.temperature
        state.maxTurns = settings.maxTurns
        state.autoApproveTools = settings.autoApproveTools
        state.useDarkTheme = settings.useDarkTheme
    }
    
    /**
     * 重置为默认值
     */
    fun resetToDefaults() {
        state = State()
    }
    
    /**
     * 获取 API Key
     */
    fun getApiKey(): String? {
        return state.apiKey
    }
    
    /**
     * 设置 API Key
     */
    fun setApiKey(apiKey: String?) {
        state.apiKey = apiKey
    }
    
    override fun getState(): State {
        return state
    }
    
    override fun loadState(state: State) {
        this.state = state
    }
    
    companion object {
        fun getInstance(project: Project): ClaudeSettingsService {
            return project.getService(ClaudeSettingsService::class.java)
        }
    }
}

File: vscode-extension/src/ide/services/claudeSettingsService.ts
/**
 * Claude Settings Service
 * 
 * Manages Claude-related configuration (API Key, model, etc.)
 * Translated from jetbrains-plugin/src/main/kotlin/com/asakii/plugin/services/ClaudeSettingsService.kt
 */

import * as vscode from 'vscode';

/**
 * Claude settings state
 */
export interface ClaudeSettings {
    apiKey?: string;
    model: string;
    maxTokens: number;
    temperature: number;
    maxTurns: number;
    autoApproveTools: boolean;
    useDarkTheme: boolean;
}

const DEFAULT_SETTINGS: ClaudeSettings = {
    apiKey: undefined,
    model: 'claude-sonnet-4-5-20250929',
    maxTokens: 4096,
    temperature: 0.7,
    maxTurns: 10,
    autoApproveTools: false,
    useDarkTheme: true
};

const CONFIG_KEY = 'claudeCodePlus';

/**
 * Claude Settings Service
 * 
 * Uses VS Code workspace configuration for persistence
 */
export class ClaudeSettingsService {
    private static instance: ClaudeSettingsService | undefined;
    private changeListeners: Set<(settings: ClaudeSettings) => void> = new Set();
    
    private constructor() {}
    
    static getInstance(): ClaudeSettingsService {
        if (!this.instance) {
            this.instance = new ClaudeSettingsService();
        }
        return this.instance;
    }
    
    /**
     * Get current settings
     */
    getSettings(): ClaudeSettings {
        const config = vscode.workspace.getConfiguration(CONFIG_KEY);
        return {
            apiKey: config.get('apiKey', DEFAULT_SETTINGS.apiKey),
            model: config.get('model', DEFAULT_SETTINGS.model),
            maxTokens: config.get('maxTokens', DEFAULT_SETTINGS.maxTokens),
            temperature: config.get('temperature', DEFAULT_SETTINGS.temperature),
            maxTurns: config.get('maxTurns', DEFAULT_SETTINGS.maxTurns),
            autoApproveTools: config.get('autoApproveTools', DEFAULT_SETTINGS.autoApproveTools),
            useDarkTheme: config.get('useDarkTheme', DEFAULT_SETTINGS.useDarkTheme)
        };
    }
    
    /**
     * Update settings
     */
    async updateSettings(settings: Partial<ClaudeSettings>): Promise<void> {
        const config = vscode.workspace.getConfiguration(CONFIG_KEY);
        
        for (const [key, value] of Object.entries(settings)) {
            if (value !== undefined) {
                await config.update(key, value, vscode.ConfigurationTarget.Workspace);
            }
        }
        
        this.notifyChange();
    }
    
    /**
     * Reset to defaults
     */
    async resetToDefaults(): Promise<void> {
        await this.updateSettings(DEFAULT_SETTINGS);
    }
    
    /**
     * Get API Key
     */
    getApiKey(): string | undefined {
        return this.getSettings().apiKey;
    }
    
    /**
     * Set API Key
     */
    async setApiKey(apiKey: string | undefined): Promise<void> {
        await this.updateSettings({ apiKey });
    }
    
    /**
     * Get model
     */
    getModel(): string {
        return this.getSettings().model;
    }
    
    /**
     * Set model
     */
    async setModel(model: string): Promise<void> {
        await this.updateSettings({ model });
    }
    
    /**
     * Add change listener
     */
    addChangeListener(listener: (settings: ClaudeSettings) => void): void {
        this.changeListeners.add(listener);
    }
    
    /**
     * Remove change listener
     */
    removeChangeListener(listener: (settings: ClaudeSettings) => void): void {
        this.changeListeners.delete(listener);
    }
    
    /**
     * Notify all listeners
     */
    private notifyChange(): void {
        const settings = this.getSettings();
        this.changeListeners.forEach(listener => listener(settings));
    }
}

// Export singleton instance
export const claudeSettingsService = ClaudeSettingsService.getInstance();
